code/datums/traits/negative.dm

```diff
@@ -559,6 +559,65 @@
 			return
 		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "wrong_cigs", /datum/mood_event/wrong_brand)
 
+/datum/quirk/alcoholic
+	name = "Alcoholic"
+	desc = "You can't stand being sober."
+	value = -1
+	gain_text = "<span class='danger'>You could really go for a drink right about now.</span>"
+	lose_text = "<span class='notice'>You feel like you should quit drinking.</span>"
+	medical_record_text = "Patient is an alcohol abuser."
+	var/where_drink //Where the bottle spawned
+	var/drink_types = list(/obj/item/reagent_containers/food/drinks/bottle/ale,
+					/obj/item/reagent_containers/food/drinks/bottle/beer,
+					/obj/item/reagent_containers/food/drinks/bottle/gin,
+		            /obj/item/reagent_containers/food/drinks/bottle/whiskey,
+					/obj/item/reagent_containers/food/drinks/bottle/vodka,
+					/obj/item/reagent_containers/food/drinks/bottle/rum,
+					/obj/item/reagent_containers/food/drinks/bottle/applejack)
+	var/need = 0 // How much they crave alcohol at the moment
+	var/tick_number = 0 // Keeping track of how many ticks have passed between a check
+	var/obj/item/reagent_containers/food/drinks/bottle/drink_instance
+
+/datum/quirk/alcoholic/on_spawn()
+	drink_instance = pick(drink_types)
+	drink_instance = new drink_instance()
+	var/list/slots = list("in your backpack" = SLOT_IN_BACKPACK)
+	var/mob/living/carbon/human/H = quirk_holder
+	where_drink = H.equip_in_one_of_slots(drink_instance, slots, FALSE) || "at your feet"
+
+/datum/quirk/alcoholic/post_add()
+	to_chat(quirk_holder, "<span class='boldnotice'>There is a small bottle of [drink_instance] [where_drink]. You only have a single bottle, might have to find some more...</span>")
+
+datum/quirk/alcoholic/on_process()
+	if(tick_number >= 6) // how many ticks should pass between a check
+		tick_number = 0
+		var/mob/living/carbon/human/H = quirk_holder
+		if(H.drunkenness > 0) // If they're not drunk, need goes up. else they're satisfied
+			need = -15
+		else
+			need++
+
+		switch(need)
+			if(1 to 10)
+				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "alcoholic", /datum/mood_event/withdrawal_light, "alcohol")
+				if(prob(5))
+					to_chat(H, "<span class='notice'>You could go for a drink right about now.</span>")
+			if(10 to 20)
+				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "alcoholic", /datum/mood_event/withdrawal_medium, "alcohol")
+				if(prob(5))
+					to_chat(H, "<span class='notice'>You feel like you need alcohol. You just can't stand being sober.</span>")
+			if(20 to 30)
+				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "alcoholic", /datum/mood_event/withdrawal_severe, "alcohol")
+				if(prob(5))
+					to_chat(H, "<span class='danger'>You have an intense craving for a drink.</span>")
+			if(30 to INFINITY)
+				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "alcoholic", /datum/mood_event/withdrawal_critical, "Alcohol")
```

Comment:
The alcohol is capitalized on this one, but not all the others. Not sure if that affects anything in game. Also the "alcoholic" mood event source should be a define for consistency

Reply:
the alcohol is capitalized on this one cuz the message is "[input]! [input]! [input]!" so it doesn't look right if it's not capitalized

Reply:
and we talked about this on discord, no other mood even sources I can see have defines